crates/edr_evm/src/block/builder.rs

```diff
@@ -110,6 +110,7 @@ where
     fn add_transaction(
         &mut self,
         transaction: ChainSpecT::SignedTransaction,
+        custom_precompiles: &HashMap<Address, PrecompileFn>,
```

Comment:
Currently it's possible for different transactions to be mined with different precompiles. Why did you not add them at construction time of the `BlockBuilder`?

Reply:
That's a good suggestion for improvement. No need to phrase it like this though.